### DISCUSSION TITLE
Which machine.conf file for Orin Nano 8GB?

### DISCUSSION THREAD

**Discussion by fre-wolf** [OE4T/meta-tegra#1584] [Q&A] [answered]
Hello everyone,

I have the Jetson Orin Nano 8GB (900-13767-0030-000) and want to know if it's safe to use the p3768-0000-p3767-0004.conf (intended for the 4GB variant) file. 

Will I have all cuda cores available? Is the p3768-0000-p3767-0001.conf from the Orin NX 8GB a better choice?

Thank you,
Frederik

**Comment by madisongh** [comment, marked as answer]
I think you meant `0003` instead of `0030` in that board SKU, right?  I think you'd be better off starting with `jetson-orin-nano-devkit` as your base - the only difference that I know of between the module in the stock devkit (sku `0005`) and the regular 8GB module (sku `0003`) is the presence of the SDcard slot of the devkit module. There appear to be more differences between the 8GB SKU and the 4GB one.

So creating a `p3768-0000-p3767-0003` config that modifies the storage layout setting and, I think, the device tree setting before including the `orin-nano.inc` common config, is probably the right approach.

**Reply by fre-wolf** [reply]
if I want to adjust the storage layout, can I just create my own `flash_l4t_t234_usb_stick.xml` based on `flash_l4t_t234_nvme.xml` and set `PARTITION_LAYOUT_EXTERNAL_DEFAULT` to that? If yes, is it enough to put the new xml in `./conf/machine` or how could I use that?
Does that way allow me to limit the APP partition to 9GB while having a rootfs around 8GB or what is best practice?

**Reply by madisongh** [reply]
Sure, you can create your own storage layout and have the variable use that name. Adding a bbappend for the `tegra-storage-layout` recipe is the main way to get your custom layout installed for use... with scarthgap and later. If you're still on the kirkstone branch, it's a little more involved - add a recipe to stage your custom layout, then add a bbappend for the `tegra-bootfiles` recipe to get it installed in the right place.

If you only need to modify the APP partition size, though, you might still use the NVIDIA default storage layout, then set TEGRA_EXTERNAL_DEVICE_SECTORS to control the total size of the storage device, ROOTFSPART_SIZE to control the size of the APP partition, and the regular Yocto IMAGE_ROOTFS_* variables to control the size of the root filesystem.
